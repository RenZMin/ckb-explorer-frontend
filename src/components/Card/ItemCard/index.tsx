import { ReactNode } from 'react'
import { ItemCardPanel, ItemContentPanel, ItemDetailPanel } from './styled'

export interface ItemCardData {
  title: string
  content: ReactNode
}

export const ItemDetail = ({ item, hideLine }: { item: ItemCardData; hideLine: boolean }) => (
  <ItemDetailPanel hideLine={hideLine}>
    <div className="item__detail__title">{item.title}</div>
    <div className="item__detail__value">{item.content}</div>
  </ItemDetailPanel>
)

export default ({
  items,
  children,
  outputIndex,
}: {
  items: ItemCardData[]
  children?: ReactNode
  outputIndex?: string
}) => (
  <ItemCardPanel id={outputIndex ? `output_${outputIndex}` : ''}>
    <ItemContentPanel>
      {items.map((item, index) => (
        <ItemDetail key={item.title} item={item} hideLine={index === items.length - 1} />
      ))}
    </ItemContentPanel>
    {children}
  </ItemCardPanel>
)
