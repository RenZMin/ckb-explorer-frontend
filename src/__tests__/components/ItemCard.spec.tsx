import { ReactElement } from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'
import ItemCard, { ItemDetail } from '../../components/Card/ItemCard'

describe('ItemCard Component', () => {
  let component: ReactElement

  beforeAll(() => {
    const items = [
      {
        title: 'title1',
        content: 'content1',
      },
      {
        title: 'title2',
        content: <div>content2</div>,
      },
      {
        title: 'title3',
        content: 'content3',
      },
      {
        title: 'title4',
        content: <div>content4</div>,
      },
    ]
    component = <ItemCard items={items} />
  })

  it('shallow renders', () => {
    const wrapper = renderer.create(component).toJSON()
    expect(wrapper).toMatchSnapshot()
  })

  it('Component Render', () => {
    const wrapper = shallow(component)
    expect(wrapper).toBeDefined()
    expect(wrapper.find(ItemDetail)).toHaveLength(4)
  })
})
