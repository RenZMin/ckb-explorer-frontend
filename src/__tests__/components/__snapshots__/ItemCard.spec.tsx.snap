// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemCard Component shallow renders 1`] = `
<div
  className="sc-bdnxRM dlIANJ"
  id=""
>
  <div
    className="sc-gtsrHT fA-dmiC"
  >
    <div
      className="sc-dlnjwi hfZqzQ"
    >
      <div
        className="item__detail__title"
      >
        title1
      </div>
      <div
        className="item__detail__value"
      >
        content1
      </div>
    </div>
    <div
      className="sc-dlnjwi hfZqzQ"
    >
      <div
        className="item__detail__title"
      >
        title2
      </div>
      <div
        className="item__detail__value"
      >
        <div>
          content2
        </div>
      </div>
    </div>
    <div
      className="sc-dlnjwi hfZqzQ"
    >
      <div
        className="item__detail__title"
      >
        title3
      </div>
      <div
        className="item__detail__value"
      >
        content3
      </div>
    </div>
    <div
      className="sc-dlnjwi kGkqrj"
    >
      <div
        className="item__detail__title"
      >
        title4
      </div>
      <div
        className="item__detail__value"
      >
        <div>
          content4
        </div>
      </div>
    </div>
  </div>
</div>
`;
